README.md

```diff
@@ -2,30 +2,384 @@
-appropriate to help with the reader's orientation with the material and future
-referencing.
+`long	count()`
 
-## Topic 2
+The `count` method returns a `long` representing the number of elements in the stream.
 
-Discuss the topic of the reading. Break into sections and sub-sections as
-appropriate to help with the reader's orientation with the material and future
-referencing.
+```java
+import java.util.List;
+import java.util.stream.Stream;
 
-## Conclusion
+public class Example {
+    public static void main(String[] args) {
+        List<Integer> numbers = List.of(-50, 20, 12, 4, -9);
+        long numCount = numbers.stream().count();
+        System.out.println(numCount); //5
 
-A short one or two paragraph summary of the contents of the lessons, recapping
-the learning goals.
+        Stream<String> greetStream = Stream.of("hello", "hi there");
+        System.out.println(greetStream.count()); //2
+    }
+}
+```
+
+## max and min
+
+`Optional<T>	max(Comparator<? super T> comparator)`  
+`Optional<T>	min(Comparator<? super T> comparator)`
+
+The `max` method takes a `Comparator` as a parameter and returns a `Optional<T>` representing
+the maximum element of the stream according to the provided Comparator.
+
+- `T` is the type of the elements in the stream's data source.
+- The `Comparator` corresponds to the data source type `T`, i.e. `Integer::compare` for an `Integer` array or list.
+- The return type is `Optional` since the operation could be called on an empty collection, in which case there is no max value.
+
+In the example below,  the `max` function returns the 
+maximum value `55` wrapped in an `Optional` object
+when called on a non-empty list `ages`.
+The `max` function returns `Optional.empty` when
+called on the empty list `prices`.
+
+```java
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class Example {
+  public static void main(String[] args) {
+    List<Integer> ages = List.of(12, 55, 37, 9);
+    //List is not empty, a maximum value is returned as Optional[55]
+    Optional<Integer> maxNumber = ages.stream().max(Integer::compare);
+    System.out.println(maxNumber);   //Optional[55]
+    maxNumber.ifPresentOrElse(System.out::println,
+            () -> System.out.println("max age unavailable"));
+
+    List<Double> prices = new ArrayList<Double>();
+    //List is empty, max method returns Optional.empty
+    Optional<Double> maxPrice = prices.stream().max(Double::compare);
+    System.out.println(maxPrice);    //Optional.empty
+    maxPrice.ifPresentOrElse(System.out::println,
+            () -> System.out.println("max price unavailable"));
+  }
+}
+```
+
+The program prints:
+
+```text
+Optional[55]
+55
+Optional.empty
+max price unavailable
+```
+
+The `min` method is similar in that it takes a `Comparator` as
+parameter and returns an `Optional`, the minimum element in the stream.
+
+
+### Primitive stream max and min methods
```

Comment:
Not sure how often we even find ourselves using the min and max methods, but would we want to showcase using it with a data type that isn't as intuitive than a number? Like a LocalDate or even String?

Reply:
Good suggestion.  I added an example with an Employee class and comparators for age and salary to find min and max.  Does that work?